modules/lotamePanoramaIdSystem.js

```diff
@@ -14,10 +14,41 @@ const KEY_EXPIRY = `${KEY_ID}_expiry`;
 const KEY_PROFILE = '_cc_id';
 const MODULE_NAME = 'lotamePanoramaId';
 const NINE_MONTHS_MS = 23328000 * 1000;
-const DAYS_TO_CACHE = 7;
-const DAY_MS = 60 * 60 * 24 * 1000;
 
 export const storage = getStorageManager(null, MODULE_NAME);
+const cookieDomain = findCookieDomain(String(document.domain));
+
+/**
+ * Find the root dommain
+ * @param {string} fullDomain
+ * @return {?string}
+ */
+function findCookieDomain(fullDomain) {
+  const domainParts = fullDomain.split('.');
+  if (domainParts.length == 2) {
+    return fullDomain;
+  }
+  let rootDomain;
+  let continueSearching = true;
+  let startIndex = -2;
+  const TEST_COOKIE_NAME = 'lotame_domain_check';
+  const TEST_COOKIE_VALUE = 'writeable';
+  do {
```

Comment:
Have we sanity checked this on a couple different browsers?

Reply:
Checked with firefox and then needed to refactor...